Validation/HGCalValidation/plugins/HGCGeometryValidation.cc

```diff
@@ -295,9 +297,10 @@ void HGCGeometryValidation::analyze(const edm::Event &iEvent,
 	HcalCellType::HcalCell cell = hcons_->cell(subdet, zside, lay, eta, phi);
 	
 	double zp = cell.rz/10; //mm --> cm
+	if (zside == 0) zp = -zp;
 	double rho = zp*TMath::Tan(2.0*TMath::ATan(TMath::Exp(-cell.eta)));
-	double xp = rho * TMath::Cos(cell.phi); //cm
-	double yp = rho * TMath::Sin(cell.phi); //cm
+	double xp  = rho * TMath::Cos(cell.phi); //cm
```

Comment:
I know its not new code- but please change to native C++ cos sin here. Thx @bsunanda